### DISCUSSION TITLE
Authorization [CLIENT_CREDENTIALS]

### DISCUSSION THREAD

**Discussion by georgev-97** [mia-platform/community#508] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12

### Description

Hello,

is it possible to secure and endpoint exposed by the api-gateway with client id and client secret?

**Comment by fredmaggiowski** [comment, marked as answer]
Hi @georgev-97 sorry for the late reply!

Yes, it is possible, however this can be done only via advanced configuration.

The configuration to be used changes depending on which API Gateway is being installed in your project (the nginx-based or envoy-based).

For example, in Envoy you could use [JWT Authentication filter](https://docs.mia-platform.eu/docs/development_suite/api-console/advanced-section/api-gateway-envoy/jwt-authn) to reach the client-credentials service and unpack the token for validation.

Is this still relevant to your use-case?
